website/static/js/filepage/revisions.js

```diff
@@ -58,8 +58,8 @@ var FileRevisionsTable = {
                 if (model.selectedRevision === 0) {
                     self.enableEditing();
                 }
-                model.hasUser = model.revisions[0] && model.revisions[0].extra && model.revisions[0].extra.user;
-                model.hasHashes = model.revisions && model.revisions[0] && model.revisions[0].extra.hashes;
+                model.hasUser = model.revisions[0] && model.revisions[0].attributes.extra && model.revisions[0].attributes.extra.user;
```

Comment:
Instead of adding `attributes` to the dereference chain throughout the code, can you extract the attributes when `model.revisions` is created?